src/Function.h

```diff
@@ -75,6 +77,10 @@ class Function {
     /** Construct a Function from an existing FunctionContents pointer. Must be non-null */
     EXPORT explicit Function(const IntrusivePtr<FunctionContents> &);
 
+    /** Deep copy this Function into 'copy'. Note: this method does not deep-copy
+     * the Parameter objects. */
```

Comment:
Please expand this comment a little to list what *is* copied. Recursively deep copies all called functions, reduction domains, schedules, ...